File: bottle_filler/bottle_filler/api/sinvoice.js
// Copyright (c) 2024 Adesina Akinyemi and contributors
// For license information, please see license.txt

frappe.ui.form.on("Sales Invoice Item", "empty_bottle_qty", function(frm, cdt, cdn) {
    var d = locals[cdt][cdn];
    frappe.call({
        method: "bottle_filler.bottle_filler.api.sinvoice.get_item_price",
        args: {
            item_code: d.empty_bottle_item_code,
            price_list: "Standard Buying"
        },
        callback: function(r) {
            if (r.message) {
                d.empty_bottle_rate = r.message.price_list_rate;
                d.empty_bottle_amount = r.message.price_list_rate * d.empty_bottle_qty;
                refresh_field("empty_bottle_rate", d.name);
                refresh_field("empty_bottle_amount", d.name);
            }
        }
    });
});

File: bottle_filler/bottle_filler/api/sinvoice.py
# -*- coding: utf-8 -*-
# Copyright (c) 2024, Adesina Akinyemi and contributors
# For license information, please see license.txt


from __future__ import unicode_literals
import frappe
from frappe import _
from frappe.utils import flt, add_days


@frappe.whitelist()
def get_item_price(item_code, price_list):
    price = frappe.db.get_value("Item Price", {"item_code": item_code, "price_list": price_list}, "price_list_rate")
    return {"price_list_rate": price}